### PR TITLE
Java module

### PR DESCRIPTION
**Pull request by Chang-CH**
# Description

Includes [java-packages](https://github.com/source-academy/java-packages) as submodule.

It exports an object of type `{[key: string]: string}`

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by leeyi45** [comment]
Perhaps to futureproof this somewhat, should we have a different folder under `src` for packages like these? @martin-henz @RichDom2185

**Comment by martin-henz** [comment]
> Perhaps to futureproof this somewhat, should we have a different folder under `src` for packages like these? @martin-henz @RichDom2185

In js-slang, we have `src/alt-langs`, and `scm-slang` is a submodule there:
`js-slang/src/alt-langs/scheme/scm-slang/...`
 so how about this:
 `modules/src/alt-langs/java/java-standard-packages/src/...json`

**Comment by martin-henz** [comment]
> > Perhaps to futureproof this somewhat, should we have a different folder under `src` for packages like these? @martin-henz @RichDom2185
> 
> In js-slang, we have `src/alt-langs`, and `scm-slang` is a submodule there: `js-slang/src/alt-langs/scheme/scm-slang/...` so how about this: `modules/src/alt-langs/java/java-packages/src/...json`
> 
> (edited above: java-standard-packages => java-packages. Reason: The naming scheme already provides for enough structure. "Non-standard" packages would fit into java-packages by just using appropriate qualified names.)

OK, at the moment @Chang-CH has:
```
[submodule "src/java/java-packages"]
	path = src/java/java-packages
```
This seems reasonable.

**Review by martin-henz — APPROVED** [review]
LGTM

File: .gitmodules
Comment (RichDom2185) [inline .gitmodules:1]: Why is the same submodule updated in two paths?

**Pull request merged**